File: src/iOSXamarinBenchmark/iOSXamarinBenchmark/ViewControllers/CameraAccessViewController.cs
using Foundation;
using System;
using System.CodeDom.Compiler;
using UIKit;

namespace iOSXamarinBenchmark
{
	partial class CameraAccessViewController : UIViewController
	{
		private UIImagePickerController picker { get; set; }

		public CameraAccessViewController (IntPtr handle) : base (handle)
		{
			
		}

		public override void ViewDidLoad ()
		{
			buttonTakePhoto.TouchUpInside += OpenCameraViewController;
		}

		public void OpenCameraViewController (object sender, EventArgs ea) {

			picker = new UIImagePickerController();
			picker.AllowsEditing = true;

			if(UIImagePickerController.IsSourceTypeAvailable(UIImagePickerControllerSourceType.Camera))
			{
				picker.SourceType = UIImagePickerControllerSourceType.Camera;
			}
			else if(UIImagePickerController.IsSourceTypeAvailable(UIImagePickerControllerSourceType.PhotoLibrary))
			{
				picker.SourceType = UIImagePickerControllerSourceType.PhotoLibrary;
			}

			picker.FinishedPickingMedia += (s, e) => {
				UIImage editedImage = e.Info[UIImagePickerController.EditedImage] as UIImage;
				imageViewPhoto.Image = editedImage;
	
				picker.DismissViewController (true, null);
			};

			picker.Canceled  += (s, e) => {
				picker.DismissViewController(true, null);
			};

			PresentViewController(picker, true, null);
		}
	}
}

File: src/iOSNativeBenchmark/iOSNativeBenchmark/ViewControllers/CameraAccess/CameraAccessViewController.m
//
//  CameraAccessViewController.m
//  test2
//
//  Created by pedro on 27/06/15.
//  Copyright (c) 2015 RP. All rights reserved.
//

#import "CameraAccessViewController.h"

@interface CameraAccessViewController ()

@property (weak, nonatomic) IBOutlet UIImageView *imageViewPhoto;
@property (weak, nonatomic) IBOutlet UIButton *buttonTakePhoto;

@end

@implementation CameraAccessViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)buttonTakePhotoTouched:(id)sender {
    UIImagePickerController *picker = [[UIImagePickerController alloc] init];
    picker.delegate = self;
    picker.allowsEditing = YES;
    
    if([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera])
    {
        picker.sourceType = UIImagePickerControllerSourceTypeCamera;
    }
    else if([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypePhotoLibrary])
    {
        picker.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
    }
    
    [self presentViewController:picker animated:YES completion:NULL];
}

- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary *)info {
    UIImage *chosenImage = info[UIImagePickerControllerEditedImage];
    self.imageViewPhoto.image = chosenImage;
    
    [picker dismissViewControllerAnimated:YES completion:NULL];
}

- (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker {
    [picker dismissViewControllerAnimated:YES completion:NULL];
}

@end